### PR TITLE
Add name to logs

### PR DESCRIPTION
**Pull request by olensmar**

### CODE REVIEW

File: modules/core/src/main/java/com/smartbear/readyapi4j/support/ExecutionLogger.java [hunk 1]
```diff
@@ -14,14 +17,16 @@
...
+
 /**
  * ExecutionListener that writes response HAR entries to a single log file after execution
  */
 public class ExecutionLogger implements ExecutionListener {
 
     private final static Logger LOG = LoggerFactory.getLogger(ExecutionLogger.class);
 
-    public static final String DEFAULT_EXTENSION = "log";
+    public static final String DEFAULT_EXTENSION = "json";
```
Comment (spindelmanne) [inline modules/core/src/main/java/com/smartbear/readyapi4j/support/ExecutionLogger.java]: I can see why we want something less generic than .log, but maybe .har would be better? I'm not sure I think so myself, just asking ...

File: modules/core/src/main/java/com/smartbear/readyapi4j/support/ExecutionLogger.java [hunk 2]
```diff
@@ -57,6 +71,21 @@ public void executionFinished(Execution execution) {
         }
     }
 
+    private String getExecutionName(Execution execution) {
```
Comment (spindelmanne) [inline modules/core/src/main/java/com/smartbear/readyapi4j/support/ExecutionLogger.java]: make- or create- seems more logical than get- .

File: modules/core/src/main/java/com/smartbear/readyapi4j/support/RecipeLogger.java [hunk 3]
```diff
@@ -40,12 +43,44 @@ public void filterRecipe(TestRecipe testRecipe) {
...
+                fileOutputStream.write(prettyPrintRecipe(testRecipe).getBytes());
             }
         } catch (Exception e) {
             LOG.error("Failed to write recipe to file", e);
         }
     }
+
+    private String prettyPrintRecipe(TestRecipe testRecipe) {
+        try {
+            return Json.pretty( Json.mapper().readTree(testRecipe.toString()));
```
Comment (spindelmanne) [inline modules/core/src/main/java/com/smartbear/readyapi4j/support/RecipeLogger.java]: Whitespace in brackets here and in many other places. We should reformat the code at some point.

File: modules/core/src/main/java/com/smartbear/readyapi4j/support/RecipeLogger.java [hunk 4]
```diff
@@ -40,12 +43,44 @@ public void filterRecipe(TestRecipe testRecipe) {
...
+        } catch (IOException e) {
+            return testRecipe.toString();
+        }
+    }
+
+    public static String createFileName(String str, char whitespaceChar) {
+        StringBuilder result = new StringBuilder();
```
Comment (spindelmanne) [inline modules/core/src/main/java/com/smartbear/readyapi4j/support/RecipeLogger.java]: If you accept leaving the 0 char in the string, you could replace the whole method with return str.replaceAll("\\s", String.valueOf(whitespaceChar))

File: modules/facade/src/main/java/com/smartbear/readyapi4j/facade/execution/RecipeExecutionFacade.java [hunk 5]
```diff
@@ -33,7 +33,24 @@
 
     public static RecipeExecutionResult executeRecipe(TestStepBuilder... testStepBuilders) {
         TestRecipe recipe = TestRecipeBuilder.newTestRecipe(testStepBuilders).buildTestRecipe();
+        return execute(recipe);
+    }
+
+    /**
+     * Builds and executes a TestRecipe from the specified TestStepBuilders using either a local or remote
+     * executor as configured. The recipe is always executed synchronously.
```
Comment (spindelmanne) [inline modules/facade/src/main/java/com/smartbear/readyapi4j/facade/execution/RecipeExecutionFacade.java]: Good Javadoc. Do we need a method for async execution too? I suppose that won't work so well with local execution though?

File: modules/core/src/main/java/com/smartbear/readyapi4j/support/ExecutionLogger.java [hunk 6]
```diff
@@ -48,7 +53,16 @@ public void executionFinished(Execution execution) {
...
             }
 
-            File file = File.createTempFile( "execution-" + execution.getId(), "." + extension, directory);
+            File file;
+            String name = getExecutionName(execution);
+
+            if(StringUtils.isNotBlank(name)){
+                file = new File( directory, createFileName( name, '_') + "." + extension );
+            }
+            else {
```
Comment (spindelmanne) [inline modules/core/src/main/java/com/smartbear/readyapi4j/support/ExecutionLogger.java]: Formatting

File: modules/testserver/src/main/java/com/smartbear/readyapi4j/testserver/execution/TestServerRecipeExecutor.java [hunk 7]
```diff
@@ -58,17 +57,17 @@ public TestServerExecution executeRecipe(TestRecipe recipe) {
...
 
-    private TestServerExecution doExecuteTestCase(TestCase testCase, ExtractorData optionalExtractorData, boolean async)  {
+    private TestServerExecution doExecuteTestCase(TestRecipe testRecipe, ExtractorData optionalExtractorData, boolean async)  {
         try {
             Optional<ExtractorData> extractorDataOptional = Optional.ofNullable(optionalExtractorData);
             extractorDataOptional.ifPresent(extractorData -> extractorDataList.add(extractorData));
-            TestServerExecution execution = testServerClient.postTestRecipe(testCase, async);
-            cancelExecutionAndThrowExceptionIfPendingDueToMissingClientCertificate(execution.getCurrentReport(), testCase);
+            TestServerExecution execution = testServerClient.postTestRecipe(testRecipe, async);
+            cancelExecutionAndThrowExceptionIfPendingDueToMissingClientCertificate(execution.getCurrentReport(), testRecipe.getTestCase());
```
Comment (spindelmanne) [inline modules/testserver/src/main/java/com/smartbear/readyapi4j/testserver/execution/TestServerRecipeExecutor.java]: That method name is a mouthful ... but maybe it's a good one. :-) And it's not new anyway.

File: modules/core/src/main/java/com/smartbear/readyapi4j/support/ExecutionLogger.java (hunk 1 above)
Comment (olensmar) [inline modules/core/src/main/java/com/smartbear/readyapi4j/support/ExecutionLogger.java]: I choose json to get syntax-highlighting in IDEA... I agree that .har sounds better

File: modules/core/src/main/java/com/smartbear/readyapi4j/support/ExecutionLogger.java [hunk 8]
```diff
@@ -57,6 +71,21 @@ public void executionFinished(Execution execution) {
...
 
+    private String createExecutionName(Execution execution) {
+        ProjectResultReport currentReport = execution.getCurrentReport();
+        String name = currentReport.getProjectName();
+        List<TestSuiteResultReport> testSuiteResultReports = currentReport.getTestSuiteResultReports();
+        if( testSuiteResultReports.size() == 1 ) {
+            if( testSuiteResultReports.get(0).getTestCaseResultReports().size()==1) {
+                name = "TestCase " + testSuiteResultReports.get(0).getTestCaseResultReports().get(0).getTestCaseName();
+            }
+            else {
```
Comment (prakashmrec) [inline modules/core/src/main/java/com/smartbear/readyapi4j/support/ExecutionLogger.java]: formatting

**Comment by spindelmanne** [comment]
I'm fine with merging this. Any comments, @prakashmrec ?

**Comment by spindelmanne** [comment]
Just resolve the conflicts and we will merge!

**Comment by olensmar** [comment]
@spindelmanne done!

**Pull request merged**